### PR TITLE
add a .merlin target to the makefile

### PR DESCRIPTION
**Pull request by gmalecha**
This is helpful when writing Coq plugins.

### CODE REVIEW

File: tools/coq_makefile.ml [hunk 1]
```diff
@@ -800,6 +806,21 @@ let check_overlapping_include (_,inc_i,inc_r) =
...
+ *)
+let merlin targets (ml_inc,_,_) =
+  print ".merlin:\n";
+  print "\t@echo 'FLG -rectypes' > .merlin\n" ;
+  List.iter (fun c ->
+      print "\t@echo \"B $(COQLIB)" ; print c ; print "\" >> .merlin\n")
+    lib_dirs ;
+  List.iter (fun (_,c) ->
+      print "\t@echo \"B " ; print c ; print "\" >> .merlin\n" ;
+      print "\t@echo \"S " ; print c ; print "\" >> .merlin\n")
```
Comment (gasche) [inline tools/coq_makefile.ml]: You could use `printf "\t@echo \"B %s\" >> .merlin\n" c` for readability (same thing for other sequenced `print` calls.

**Comment by gasche** [comment]
This seems to be an excellent idea.

I reviewed the part of the patch that factorizes existing code to make sure there is no regression. The patch is safe and could be applied.

File: tools/coq_makefile.ml (hunk 1 above)
Comment (gmalecha) [inline tools/coq_makefile.ml]: I didn't do this because most of the rest of the file does not do this. I'll push an update shortly.

**Comment by maximedenes** [comment]
Merged in 8.5 branch, thanks a lot!

**Pull request closed without merging**